src/LtiLibrary.AspNetCore/Outcomes/v1/OutcomesControllerBase.cs

```diff
@@ -223,10 +228,21 @@ private static Result GetResult(ResultRecordType resultRecord)
                         response.StatusCode, response.StatusDescription ?? "TC did not read result.");
                 }
 
+                readResponse.result = new ResultType { resultScore = new TextType() };
+                // Add optional responseData if present
+                if (!string.IsNullOrEmpty(response.Result.LtiLaunchUrl + response.Result.Text + response.Result.Url))
+                {
+                    readResponse.result.ResultData = new DataType
+                    {
+                        LtiLaunchUrl = response.Result.LtiLaunchUrl,
+                        Text = response.Result.Text,
+                        Url = response.Result.Url
+                    };
+                }
+
                 if (response.Result.Score.HasValue)
                 {
                     // The score exists
-                    readResponse.result = new ResultType { resultScore = new TextType() };
                     var cultureInfo = new CultureInfo("en");
```

Comment:
I'm not sure but do you see any potential issue to move default `CultireInfo` instance to `LtiConstants`?